internal/services/containers/container_registry_task_resource.go

```diff
@@ -1618,3 +1627,28 @@ func flattenRegistryTaskAgentProperties(input *legacyacr.AgentProperties) []Agen
 	}
 	return []AgentConfig{{CPU: cpu}}
 }
+
+func patchRegistryTaskTriggerSourceTrigger(triggers *[]legacyacr.SourceTrigger, model ContainerRegistryTaskModel) *[]legacyacr.SourceTrigger {
+	if triggers == nil {
+		return triggers
+	}
+	if len(*triggers) != len(model.SourceTrigger) {
+		return triggers
+	}
+
+	result := make([]legacyacr.SourceTrigger, len(*triggers))
+	for i, trigger := range model.SourceTrigger {
+		t := (*triggers)[i]
+		if len(trigger.Auth) == 0 {
+			result[i] = t
+			continue
+		}
+		if t.SourceRepository == nil {
+			result[i] = t
+			continue
+		}
+		t.SourceRepository.SourceControlAuthProperties = expandRegistryTaskAuthInfo(trigger.Auth[0])
+		result[i] = t
+	}
+	return &result
+}
```

Comment:
meaning this can become:

```suggestion
func patchRegistryTaskTriggerSourceTrigger(triggers []legacyacr.SourceTrigger, model ContainerRegistryTaskModel) *[]legacyacr.SourceTrigger {
	if len(triggers) != len(model.SourceTrigger) {
		return triggers
	}

	result := make([]legacyacr.SourceTrigger, len(triggers))
	for i, trigger := range model.SourceTrigger {
		t := triggers[i]
		if len(trigger.Auth) == 0 {
			result[i] = t
			continue
		}
		if t.SourceRepository == nil {
			result[i] = t
			continue
		}
		t.SourceRepository.SourceControlAuthProperties = expandRegistryTaskAuthInfo(trigger.Auth[0])
		result[i] = t
	}
	return &result
}
```